### ISSUE TITLE
AddToCard action not throwing an consumable response if product is out-of-stock

### ISSUE THREAD

**Issue by danielkoch** [adobe/aem-core-cif-components#917]
### Expected Behaviour
If add to cart query is running against an out-of-stock product it should expose the error to the frontend. Otherwise it's impossible to implement a user interface an inform the user. 

### Actual Behaviour
Request that is trigged by the DOM event `aem.cif.add-to-cart` throw the following error:
<img width="684" alt="AddToCartNoStock" src="https://user-images.githubusercontent.com/142055/171637791-eb0ce9c4-d94b-4ffa-a619-8522d0e8045b.png">

### Reproduce Scenario (including but not limited to)
Product must be out-of-stock

#### Steps to Reproduce
1. Click on "Add to cart" on a product that is out-of-stock
2. Inspect browser console
3. Check cart amount (no changes)

#### Platform and Version
AEM v6.9
CIF components, v2.4.0
(Magento Cloud)

#### Sample Code that illustrates the problem
No example needed.

#### Logs taken while reproducing problem
Browser console see above.

**Comment by Buuhuu** [comment]
@adobe export issue to Jira project CIF

**Comment by Buuhuu** [comment]
You are right, we have a mechanism for that for the wish list. Doing the same for add to cart would certainly be useful. 

https://github.com/adobe/aem-core-cif-components/blob/master/react-components/src/talons/Wishlist/useAddToWishlistEvent.js#L46-L62

**Comment by Buuhuu** [comment]
> In addition: Also the name of the product itself
This will be added with https://github.com/adobe/aem-core-cif-components/pull/949/files#diff-d974f98a4343d70995ce10ca3ad8c97f28a5f99723ee5a63ff3d0f862f4f60af
